tests/backends/test_cursors.py

```diff
@@ -0,0 +1,55 @@
+import unittest
+
+from django.db import connection
+from django.test import TestCase
+
+
+class LoggingCursorMixin:
+    def __init__(self, bucket, *args, **kwargs):
+        super().__init__(bucket, *args, **kwargs)
+        self.bucket = bucket
+
+    def execute(self, sql, *args, **kwargs):
+        self.bucket.append(sql)
+        super().execute(sql, *args, **kwargs)
+
+
+@unittest.skipUnless(connection.vendor == 'mysql', 'MySQL specific test.')
+class MySQLCursorOptionsTestCase(TestCase):
+    try:
+        from MySQLdb.cursors import Cursor
+
+        class MySQLLoggingCursor(LoggingCursorMixin, Cursor):
+            pass
+    except ImportError:
+        pass
+
+
+@unittest.skipUnless(connection.vendor == 'oracle', 'Oracle specific test.')
+class OracleCursorOptionsTestCase(TestCase):
+    try:
+        import cx_Oracle as Database
+
+        class OracleLoggingCursor(LoggingCursorMixin, Database):
```

Comment:
I believe oracle tests need some adjustment or can be removed too. looking for @felixxm insight as I suggested most of the parts.